### DISCUSSION TITLE
How can I intercept this response - {"message": "An invalid response was received from the upstream server"}?

### DISCUSSION THREAD

**Discussion by Sereneee** [Kong/kong#7244] [Help] [answered]
I have postman, Kong and a microservice setup locally. In kong, I have a route that points to the local microservice, when it is running, I can send request through postman and receive a response perfectly fine.
When the microservice is down, I get {"message": "An invalid response was received from the upstream server"} response, which I believe is triggered because Kong detected a connection timeout (please correct me if I'm wrong).
Does anyone know how I can detect the connection timeout using a custom kong plugin or anything else?

**Comment by locao** [comment]
Hi @Sereneee! You should receive a 504 status code and the `"message"` field should contain `"The upstream server is timing out"` when the upstream is timing out. 

What do you get if you send the request directly to the microservice port when it is down?

**Comment by Sereneee** [comment]
Hi @locao , thanks for your response.

I'm getting 502 status code and the `"message"` field containing "`An invalid response was received from the upstream server"` when I send request directly to the microservice when it is down.

Is there anyway that Kong can detect a connection timeout other than receiving 502 or 504 status code and their respective messages?

**Comment by kikito** [comment, marked as answer]
This looks like a duplicate of #7242, please don't open multiple threads for the same question.
